base/math.jl

```diff
@@ -322,6 +322,10 @@ end
 # until the heuristics can be improved
 @inline nan_dom_err(out, x) = isnan(out) & !isnan(x) ? throw(DomainError(x, "NaN result for non-NaN input.")) : out
```

Comment:
I think this message should be changed too. `isnan(out) & !isnan(x)` is just an implementation trick to detect domain errors for multiple functions; it doesn't describe the actual problem.

Reply:
Yes, that was the plan with this PR. Identify the functions with known domain and replace the error messages for these to be more explicit (like for log in this PR).